File: fabric/src/main/java/nars/storage/NullInferenceRecorder.java
package nars.storage;

import nars.io.IInferenceRecorder;

/**
 * Created by jim on 1/12/16.
 */
class NullInferenceRecorder implements IInferenceRecorder {

    @Override
    public void init() {
    }

    @Override
    public void show() {
    }

    @Override
    public void play() {
    }

    @Override
    public void stop() {
    }

    @Override
    public void append(String s) {
    }

    @Override
    public void openLogFile() {
    }

    @Override
    public void closeLogFile() {
    }

    @Override
    public boolean isLogging() {
        return false;
    }
}

File: src/main/java/nars/storage/NullInferenceRecorder.kt
package nars.storage

import nars.io.IInferenceRecorder

internal class NullInferenceRecorder : IInferenceRecorder {
    override fun init() {}
    override fun show() {}
    override fun play() {}
    override fun stop() {}
    override fun append(s: String?) {}
    override fun openLogFile() {}
    override fun closeLogFile() {}
    override val isLogging: Boolean
        get() = false
}